### PR TITLE
Add options for ensureDir and ensureDirSync

### PR DESCRIPTION
**Pull request by sangdth**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/jprichardson/node-fs-extra/blob/master/docs/ensureDir-sync.md>>
- [x] Increase the version number in the header if appropriate.
- [x] Did not make substantial changes.

---

I'm new to this, please take a look and let me know if there is anything need to be done.
Thanks in advance.

### CODE REVIEW

File: types/fs-extra/fs-extra-tests.ts [hunk 1]
```diff
@@ -142,11 +145,11 @@ fs.writeJSON(file, object, writeOptions).then(() => {
 fs.writeJsonSync(file, object, writeOptions);
 fs.writeJSONSync(file, object, writeOptions);
 
-fs.ensureDir(path).then(() => {
+fs.ensureDir(path, ensureOptions).then(() => {
```
Comment (shiftkey) [inline types/fs-extra/fs-extra-tests.ts]: These tests should be additive, as I believe the previous examples are still valid.

File: types/fs-extra/fs-extra-tests.ts [hunk 2]
```diff
@@ -15,6 +15,9 @@ const modeNum = 0;
 const modeStr = "";
 const object = {};
 const errorCallback = (err: Error) => { };
+const ensureOptions: fs.EnsureOptions = {
```
Comment (shiftkey) [inline types/fs-extra/fs-extra-tests.ts]: What about tests for providing a `number` as the second parameter?

File: types/fs-extra/fs-extra-tests.ts (hunk 1 above)
Comment (sangdth) [reply types/fs-extra/fs-extra-tests.ts]: I agree, let me change it

File: types/fs-extra/fs-extra-tests.ts (hunk 2 above)
Comment (sangdth) [reply types/fs-extra/fs-extra-tests.ts]: I have added the test for number.

**Pull request merged**